pandas/_libs/tslib.pyx

```diff
@@ -725,6 +731,21 @@ cpdef array_to_datetime(ndarray[object] values, errors='raise',
         return oresult
 
 
+cdef bint _handle_error_require_iso8601(object val, int64_t* iresult,
```

Comment:
I would rather do this inline, you are mutating the result here, and it makes the logic much harder to follow. If you want to do this in a separate PR that just refactors might be ok, but not on this one.

Reply:
OK.